### PR TITLE
Update Makefile and scripts to support release/testing pipelines

### PR DESCRIPTION
**Pull request by mayankshah1607**
This PR simplifies the `make` tooling by introducing `pipenv`. Also adds necessary scripts for supporting CI pipelines that will be added via app-interface

Signed-off-by: Mayank Shah <m.shah@redhat.com>

### CODE REVIEW

**Comment by mayankshah1607** [comment]
Cc @sugarraysam

File: Makefile [hunk 1]
```diff
@@ -1,89 +1,35 @@
...
-		$(BIN)/managedtenants --environment=$${env} run --debug tasks/deploy/ ; \
-	done
-
-##@ Generators
+check:
+	pipenv run flake8 --config=$(LINTERS)/flake8 $(PY_SRCS) && \
+	pipenv run pylint --rcfile=$(LINTERS)/pylint $(PY_SRCS) && \
+	pipenv run black --config=$(LINTERS)/black --experimental-string-processing $(PY_SRCS)
+	pipenv run yamllint --config-file=$(LINTERS)/yamllint .
+	pipenv run pytest --cache-clear -v tests/
```
Comment (sugarraysam) [inline Makefile:28]: I would keep tests split from linters just because pylint takes a long time.

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+jsonschema2md = "0.2.0.post1"
+PyYAML = "5.4.1"
+pre-commit = "2.13.0"
+
+
+[packages]
+managedtenants_cli = {editable = true,path = "."}
+
+[requires]
+python_version = "3.6"
```
Comment (sugarraysam) [inline Pipfile:23]: I love pipenv especially because it enforces python version :)

File: Makefile [hunk 3]
```diff
@@ -1,89 +1,35 @@
-SHELL=/bin/bash
-.SHELLFLAGS=-euo pipefail -c
+all:
```
Comment (sugarraysam) [inline Makefile:1]: I like `make help`, but no strong feelings. This works :+1:

File: Dockerfile.test [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+WORKDIR /managedtenant-cli
+
+COPY . ./
+
+RUN pip install pipenv
+
+RUN pipenv lock --pre --clear
+
+RUN pipenv install --dev
```
Comment (sugarraysam) [inline Dockerfile.test:13]: This has been a source of pain and misery for me in the past. It hangs and fails often, so much that I would instead do:

```bash
$ pipenv lock -r > requirements.txt
$ pip install -r requirements.txt
```

File: Dockerfile.test [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+ENTRYPOINT ["make"]
+
+CMD ["check"]
```
Comment (sugarraysam) [inline Dockerfile.test:17]: If we split `make check` and `make test` lets make sure we modify this

**Review by sugarraysam — COMMENTED** [review]
Need to modify this line as well:

```makefile
SCHEMA_IN := $(PWD)/managedtenants/data/metadata.schema.yaml
MARKDOWN_OUT := $(PWD)/docs/tenants/zz_schema_generated.md
generate-md-schema: requirements-dev
	pipenv run python $(PWD)/hack/yamlschema2md.py --schema $(SCHEMA_IN) --output $(MARKDOWN_OUT)
```

I can't run `pipenv lock` for some reason. Did it work for you?

```bash
$ pipenv lock --dev
Creating a virtualenv for this project...
Pipfile: /home/sugar/opt/_mtrepos/managed-tenants-cli/Pipfile
Using /home/sugar/.pyenv/versions/3.6.14/bin/python3.6m (3.6.14) to create virtualenv...
⠸ Creating virtual environment...created virtual environment CPython3.6.14.final.0-64 in 169ms
  creator CPython3Posix(dest=/home/sugar/opt/_mtrepos/managed-tenants-cli/.venv, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/sugar/.local/share/virtualenv)
    added seed packages: pip==21.2.4, setuptools==58.0.4, wheel==0.37.0
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator

✔ Successfully created virtual environment!
Virtualenv location: /home/sugar/opt/_mtrepos/managed-tenants-cli/.venv
Locking [dev-packages] dependencies...
Building requirements...
Resolving dependencies...
✘ Locking Failed!
```

**Comment by mayankshah1607** [comment]
> I can't run pipenv lock for some reason. Did it work for you?

@sugarraysam That command won't work because one of our dependencies `black` is a pre-release. `pipenv` needs this to be handled explicitly by attaching the `--pre` flag:

```bash
pipenv lock --pre --clear
```

**Pull request merged**